src/client/utils/hooks/useSelectedInsuranceTypes.ts

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useMemo } from 'react'
+import { useLocation, useHistory } from 'react-router'
+
+// TODO: we should get this from giraffe schema
+export enum InsuranceType {
+  SWEDISH_APARTMENT = 'SWEDISH_APARTMENT',
+  SWEDISH_HOUSE = 'SWEDISH_HOUSE',
+  SWEDISH_ACCIDENT = 'SWEDISH_ACCIDENT',
+  NORWEGIAN_HOME_CONTENT = 'NORWEGIAN_HOME_CONTENT',
+  NORWEGIAN_TRAVEL = 'NORWEGIAN_TRAVEL',
+  DANISH_HOME_CONTENT = 'DANISH_HOME_CONTENT',
+  DANISH_ACCIDENT = 'DANISH_ACCIDENT',
+  DANISH_TRAVEL = 'DANISH_TRAVEL',
+}
+
+const ALL_INSURANCE_TYPES = Object.values(InsuranceType)
+const SEARCH_PARAM_NAME = 'type'
+
+const deserializeSearchParams = (searchParams: URLSearchParams) => {
+  const matches = searchParams.getAll(SEARCH_PARAM_NAME)
+  return validateInsuranceTypes(matches)
+}
+
+const validateInsuranceTypes = (
+  rawTypes: Array<string>,
+): Array<InsuranceType> =>
+  rawTypes.filter((type) =>
+    ALL_INSURANCE_TYPES.includes((type as unknown) as InsuranceType),
+  ) as Array<InsuranceType>
+
+export const useSelectedInsuranceTypes = () => {
+  const location = useLocation()
+  const history = useHistory()
+
+  const searchParams = useMemo(() => new URLSearchParams(location.search), [
+    location.search,
+  ])
+  const insuranceTypes = useMemo(() => deserializeSearchParams(searchParams), [
+    searchParams,
+  ])
+
+  const changeSelectedInsuranceTypes = useCallback(
```

Comment:
Why is `useCallback` used here? 🙃

Reply:
To give it a stable reference across re-renders. When returning a function from a hook I feel this is the safest approach so it can be used in a `useEffect` dependency array.

Reply:
OK I see your reasoning. To me it (the usage of `useCallback`) mainly raises questions though, since we don't use it in any `useEffect`s now. Isn't that something that could rather be added/changed once it's needed? Or is that risky?